src/minio.rs

```diff
@@ -308,10 +337,63 @@ impl Client {
                 resp.into_body()
                     .concat2()
                     .map_err(|err| Err::HyperErr(err))
-                    .and_then(move |chunk| b2s(chunk.into_bytes()))
+                    .and_then(move |chunk| chunk_to_string(&chunk))
                     .and_then(|s| xml::parse_bucket_list(s))
             })
     }
+
+    /// get_bucket_location - Get location for the bucket_name.
+    pub fn listen_bucket_notificaion(
+        &self,
+        bucket_name: &str,
+        prefix: Option<String>,
+        suffix: Option<String>,
+        events: Vec<String>,
+    ) -> impl Stream<Item = notification::NotificationInfo, Error = Err> {
+        // Prepare request query parameters
+        let mut query_params: HashMap<String, Vec<Option<String>>> = HashMap::new();
+        query_params.insert("prefix".to_string(), vec![prefix]);
+        query_params.insert("suffix".to_string(), vec![suffix]);
+        let opt_events: Vec<Option<String>> = events.into_iter().map(|evt| Some(evt)).collect();
+        query_params.insert("events".to_string(), opt_events);
+
+        // build signed request
+        let s3_req = S3Req {
+            method: Method::GET,
+            bucket: Some(bucket_name.to_string()),
+            object: None,
+            headers: HeaderMap::new(),
+            query: query_params,
+            body: Body::empty(),
+            ts: time::now_utc(),
+        };
+        // We'll be streaming the events out of this
+        let (tx, rx) = mpsc::unbounded_channel();
```

Comment:
Why do we need this approach? We can just return a stream from the response future directly (using maybe the flatten_stream() combinator). This approach is spawning an extra future.